internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -0,0 +1,671 @@
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: r.Namespace,
+		},
+	}
+	routeSpec := routev1.RouteSpec{
+		To: routev1.RouteTargetReference{
+			Kind:   "Service",
+			Name:   name,
+			Weight: &weight,
+		},
+		Port: &routev1.RoutePort{
+			TargetPort: intstr.FromString(name),
+		},
+		WildcardPolicy: routev1.WildcardPolicyNone,
+	}
+
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(instance, route, r.Scheme); err != nil {
+			return err
+		}
+		route.Spec = routeSpec
+		return nil
+	}
+
+	return route, mutateFn
+}
+
+func (r *AgentServiceConfigReconciler) newPostgresSecret(instance *adiiov1alpha1.AgentServiceConfig) (*corev1.Secret, controllerutil.MutateFn, error) {
+	pass, err := generatePassword()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      databaseSecretName,
+			Namespace: r.Namespace,
+		},
+		StringData: map[string]string{
+			"db.host":     "localhost",
+			"db.user":     "admin",
+			"db.password": pass,
+			"db.name":     "installer",
+			"db.port":     strconv.Itoa(int(databasePort)),
+		},
+		Type: corev1.SecretTypeOpaque,
+	}
+
+	// Only setting the owner reference to prevent clobbering the generated password.
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(instance, secret, r.Scheme); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	return secret, mutateFn, nil
+}
+
+func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(instance *adiiov1alpha1.AgentServiceConfig, host string) (*appsv1.Deployment, controllerutil.MutateFn) {
+	maxUnavailable := intstr.FromString("50%")
+	maxSurge := intstr.FromString("100%")
+
+	serviceEnv := []corev1.EnvVar{
+		{Name: "SERVICE_BASE_URL", Value: host},
+
+		// database
+		newSecretEnvVar("DB_HOST", "db.host", databaseSecretName),
+		newSecretEnvVar("DB_NAME", "db.name", databaseSecretName),
+		newSecretEnvVar("DB_PASS", "db.password", databaseSecretName),
+		newSecretEnvVar("DB_PORT", "db.port", databaseSecretName),
+		newSecretEnvVar("DB_USER", "db.user", databaseSecretName),
+
+		// image overrides
+		{Name: "AGENT_DOCKER_IMAGE", Value: AgentImage()},
+		{Name: "CONTROLLER_IMAGE", Value: ControllerImage()},
+		{Name: "INSTALLER_IMAGE", Value: InstallerImage()},
+		{Name: "SELF_VERSION", Value: ServiceImage()},
+
+		{Name: "ISO_IMAGE_TYPE", Value: "minimal-iso"},
+		{Name: "S3_USE_SSL", Value: "false"},
+		{Name: "LOG_LEVEL", Value: "info"},
+		{Name: "LOG_FORMAT", Value: "text"},
+		{Name: "INSTALL_RH_CA", Value: "false"},
+		{Name: "REGISTRY_CREDS", Value: ""},
+		{Name: "AWS_SHARED_CREDENTIALS_FILE", Value: "/etc/.aws/credentials"},
+		{Name: "DEPLOY_TARGET", Value: "k8s"},
+		{Name: "STORAGE", Value: "filesystem"},
+		{Name: "ISO_WORKSPACE_BASE_DIR", Value: "/data"},
+		{Name: "ISO_CACHE_DIR", Value: "/data/cache"},
+
+		// from configmap
+		{Name: "ADMIN_USERS", Value: "ercohen,lgamliel,mfilanov,ygoldber"},
```

Comment:
:eyes:

Reply:
Yeah, there are a few here that I don't think are relevant unless we're running in the cloud.

Reply:
What do you think of ?

```suggestion
		{Name: "ADMIN_USERS", Value: ""},
```

Reply:
Empty is already the default in the source, we can just remove the line.